modules/usrmgr/Functions/CheckValidUsername.php

```diff
@@ -9,7 +9,11 @@ function CheckValidUsername($username)
 {
     global $cfg, $db;
     if ($cfg['signon_username_unique']) {
-        $row = $db->qry_first('SELECT * FROM %prefix%user WHERE username=%string%', $username);
+        if (isset($_GET['userid'])) {
```

Comment:
Hmmm.
I would rather use $auth here to get the current user details (if he is editing his own account) and only use $_GET if we have a valid admin session (and thus are editing a different account then what is in $auth).
Your thoughts?

Reply:
As users [are not allowed to modify their usernames themselves](https://github.com/lansuite/lansuite/blob/33e5b3b6358e06fc37246c3539ae99e94f4f15ab/modules/usrmgr/add.php#L21-L25) I don't think that is an issue. It would be if that restriction won't be there though.
The `usrmgr` code also has checks to validate that `$auth['userid']` equals `$_GET['userid']` so providing a random userid as non-admin shouldn't work either.
I can modify this PR and change that code or we go with it as-is.

Reply:
No, that is fine by me then ;)